app/src/main/java/com/dsvag/currencyexchanger/data/adapters/CoinAdapter.kt

```diff
@@ -44,31 +57,36 @@ class CoinAdapter() : RecyclerView.Adapter<CoinAdapter.CoinViewHolder>() {
         notifyDataSetChanged()
     }
 
-    fun moveToTop(position: Int) {
-        data[0] = data.removeAt(    position)
-        notifyItemMoved(position, 0)
-        recyclerView.scrollToPosition(0)
-    }
-
-    inner class CoinViewHolder(private val itemBinding: RowCoinBinding) :
-        RecyclerView.ViewHolder(itemBinding.root) {
+    class CoinViewHolder(
+        private val itemBinding: RowCoinBinding,
+        private val moveToTop: (position: Int) -> Unit,
+        private val reprice: (price: Double) -> Unit,
+    ) : RecyclerView.ViewHolder(itemBinding.root) {
 
         fun bind(coin: Coin) {
             itemBinding.name.text = coin.name
             itemBinding.symbol.text = coin.symbol
-            itemBinding.price.hint = coin.quote.usd.priceForOne.toString()
+            itemBinding.price.hint = coin.quote.usd.price.toString()
             itemBinding.lastUpdate.text =
                 coin.quote.usd.lastUpdated.split("T")[1].replace(".000Z", "")
 
-            if (coin.quote.usd.priceForOne != coin.quote.usd.priceInAnotherCoin) {
-                itemBinding.price.text!!.clear()
-            } else {
+            if (coin.quote.usd.priceInAnotherCoin > 0) {
                 itemBinding.price.setText(coin.quote.usd.priceInAnotherCoin.toString())
+            } else {
+                itemBinding.price.text?.clear()
             }
 
-            itemBinding.price.setOnFocusChangeListener { view, hasFocus ->
+            itemBinding.price.setOnFocusChangeListener { _, hasFocus ->
                 if (hasFocus) {
+                    itemBinding.price.text?.clear()
+
                     moveToTop(adapterPosition)
+
+                    itemBinding.price.textChanges()
+                        .subscribeOn(Schedulers.computation())
+                        .observeOn(AndroidSchedulers.mainThread())
+                        .debounce(300, TimeUnit.MILLISECONDS)
+                        .subscribe({ reprice("0".plus(it).toDouble()) }, {}, {})
```

Comment:
`"0".plus(it)` - странно выглядит. Я бы сделал по-простому:
```suggestion
                        .subscribe({ reprice(if (it.isBlank()) 0.0 else it.toDouble()) }, {}, {})
```

Или так:
```suggestion
                        .subscribe({ reprice(it.toDoubleOrNull() ?: 0.0) }, {}, {})
```